backend/api/Controllers/UserController.cs

```diff
@@ -66,6 +75,26 @@ public async Task<IActionResult> UserInfo()
 
             return await response.HandleResponseAsync<MembershipModel.User>();
         }
+
+        /// <summary>
+        /// Allows a user to submit an access request to the system, associating a role and agency to their user.
+        /// </summary>
+        /// <returns></returns>
+        [HttpPost("/api/accessRequest")]
```

Comment:
I'm a fan of this format for routes `/api/access/request`

Reply:
sure, I'll change that.

Reply:
In the interest of keeping a RESTful approach it should be:

GET /api/access/requests  -- get many
POST /api/access/requests -- add one
PUT /api/access/requests/[:id] -- update one
DELETE /api/access/request/[:id] -- delete one

I've seen APIs where you have both the plural and singular and it gets confusing (in my opinion). Eg, /api/users and /api/user/10

Reply:
Ha I generally lean towards using the plural and singular.  But it's arbitrary, as long as we're consistent.

Reply:
Again, something to consider and it's based on preference so I'll leave to you two to decide. But I have always thought about REST endpoints as "resources" or "arrays of things" so the naming should follow that mental model (at least for me)

```
GET  /orders          <---> orders 
POST /orders          <---> orders.push(data)
GET  /orders/1        <---> orders[1]
PUT  /orders/1        <---> orders[1] = data
GET  /orders/1/lines  <---> orders[1].lines
POST /orders/1/lines  <---> orders[1].lines.push(data) 
```

By having two different endpoints /orders and /order/1 it becomes confusing when to use one or the other and I like simplicity in the sense that I don't have to remember which one to use - I always use the plural noun and let the HTTP verb (GET, POST, etc) dictate what kind of action is being taken on the "array/REST resource"

At least, that's what I've seen consistently being predicated by larger organizations & teams but again, I'll leave up to you.

https://docs.microsoft.com/en-us/azure/architecture/best-practices/api-design#define-operations-in-terms-of-http-methods
https://cloud.google.com/blog/products/api-management/restful-api-design-nouns-are-good-verbs-are-bad
https://blog.octo.com/wp-content/uploads/2014/12/OCTO-Refcard_API_Design_EN_3.0.pdf

Reply:
I'm fine with either way.  I've always leaned towards using plural and singular routes to express what they do in human readable format.

The benefit I like about your approach is you can set the default route at the top of the controller and just add the param/path parts to the endpoints.